### DISCUSSION TITLE
More support for device-service

### DISCUSSION THREAD

**Discussion by SavantEVO** [edgexfoundry/community#109] [Q&A]
Hello everybody! I have another problem that I need to know more about device service. Any built device service will have bindings to the core metadata. When a device service is built, it needs to register the device-profile in the core metadata. Please tell me more about this!
I found the device service description document in the following link
https://nexus.edgexfoundry.org/content/sites/docs/staging/california/docs/_build/html/Ch-DeviceServices.html
<img width="953" alt="image" src="https://github.com/edgexfoundry/edgex-taf/assets/109493900/6bcff7d1-211f-4ac8-aed0-d84bcc96747c">
I run docker and then check UI about the device-virtual. It has register device-profile
<img width="1268" alt="image" src="https://github.com/edgexfoundry/edgex-taf/assets/109493900/e5acd8bc-fa00-447f-ac45-98ec74ca38ee">
Thank you!

**Comment by iain-anderson** [comment]
Hi, the most recent documentation for the device service requirements is https://github.com/edgexfoundry/edgex-docs/blob/main/docs_src/design/legacy-requirements/device-service.md
Some points on this:
* It is not a complete specification, some functionality is detailed elsewhere (but linked in that doc)
* It is not updated for EdgeX v3.0
* Upload of device profiles is not addressed - this is not a requirement for device services, but the EdgeX SDKs do support upload of device profiles and devices at service startup.
